File: Unit2-06/ts/Jet.ts
/*
 * This program is teh jet class extends from the airplane
 *
 * @author Ayaat Ahmedou
 * @version 1.0
 * @since   2024-12-18
 */

import { Airplane } from "./Airplane"

/**
* This is the jet class.
*/
export class Jet extends Airplane {
    /** Jet speed change multiplier. */
    private MULTIPLIER: number = 2

    // Constructor.
    public Jet() {
        // super()
    }

    // Set jet speed.
    public setSpeed(speed: number) {
        super.setSpeed(speed * this.MULTIPLIER)
    }

    // Double speed.
    public accelerate() {
        super.setSpeed(super.getSpeed() * 2)
    }
}

File: Unit2-06/java/Jet.java
/*
* This program is the jet class.
*
* @author  Johanna Liu
* @version 1.0
* @since   2024-03-25
*/

public class Jet extends Airplane {
    /**
     * Sets the jet's speed multiplier.
    */
    private static int multiplier = 2;

    /**
     * Creates the base jet.
    */
    public Jet() {
    }

    /**
     * Sets the speed of the jet.
     *
     * @param newSpeed of the jet
    */
    public void setSpeed(int newSpeed) {
        super.setSpeed(newSpeed * Jet.multiplier);
    }

    /**
     * Accelerates the jet, changing speed.
    */
    public void accelerate() {
        super.setSpeed(this.getSpeed() * Jet.multiplier);
    }
}
